src/clients/c++/perf_client.cc

```diff
@@ -1246,15 +1242,17 @@ InferenceProfiler::SummarizeLatency(
 
   for (const auto& latency : latencies) {
     tol_latency_ns += latency;
-    tol_square_latency_us +=
-        (latency * latency) / (1000 * 1000);
+    tol_square_latency_us += (latency * latency) / (1000 * 1000);
   }
 
   summary.client_avg_latency_ns = tol_latency_ns / latencies.size();
 
   if (report_percentile_) {
-    size_t index = (percentile_ / 100.0) * latencies.size();
+    size_t index = floor((percentile_ / 100.0) * latencies.size());
```

Comment:
If percentile_ is 100.0, then index will be latencies.size(). I think you don't need the floor but you do need to use (latencies.size() - 1).